lib/internal/fs/glob.js

```diff
@@ -158,21 +158,11 @@ class Pattern {
     if (typeof pattern === 'string') {
       return pattern === path;
     }
-    if (typeof pattern?.test === 'function') {
-      return pattern.test(path);
-    }
-    return false;
+    return pattern?.test?.(path) ?? false;
   }
 
   cacheKey(index) {
-    let key = '';
-    for (let i = index; i < this.#globStrings.length; i++) {
-      key += this.#globStrings[i];
-      if (i !== this.#globStrings.length - 1) {
-        key += '/';
-      }
-    }
-    return key;
+    return this.#globStrings.slice(index).join('/');
```

Comment:
```suggestion
    return ArrayPrototypeJoin(ArrayPrototypeSlice(this.#globStrings, index), '/');
```
I am not sure about this change - it requires two iterations instead of one

Reply:
Agreed, performance-wise this won't be useful